lib/irb.rb

```diff
@@ -21,36 +21,917 @@
+# to the configuration file.
+#
+# During the session, method <tt>conf.ignore_eof?</tt> returns the setting,
+# and method <tt>conf.ignore_eof = _boolean_</tt> sets it.
+#
+# === SIGINT
+#
+# By default, <tt>IRB.conf[:IGNORE_SIGINT]</tt> is +true+,
+# which means that typing the interrupt character <tt>Ctrl-C</tt>
+# does not cause the session to exit.
+#
+# You can reverse that behavior by adding <tt>IRB.conf[:IGNORE_SIGING] = false</tt>
+# to the configuration file.
+#
+# During the session, method <tt>conf.ignore_siging?</tt> returns the setting,
+# and method <tt>conf.ignore_sigint = _boolean_</tt> sets it.
+#
+# === Automatic Completion
+#
+# By default, \IRB enables
+# {automatic completion}[https://en.wikipedia.org/wiki/Autocomplete#In_command-line_interpreters]:
+#
+# You can disable it by either of these:
+#
+# - Adding <tt>IRB.conf[:USE_AUTOCOMPLETE] = false</tt> to the configuration file.
+# - Giving command-line option <tt>--noautocomplete</tt>
+#   ()<tt>--autocomplete</tt> is the default).
+#
+# See {Notes on Initialization Precedence}[rdoc-ref:IRB@Notes+on+Initialization+Precedence].
+#
+# \Method <tt>conf.use_autocomplete?</tt> returns +true+
+# if automatic completion is enabled, +false+ otherwise.
+#
+# The setting may not be changed during the session.
+#
+# === Automatic Indentation
+#
+# By default, \IRB automatically indents lines of code to show structure
+# (e.g., it indent the contents of a block).
+#
+# The current setting is returned
+# by the configuration method <tt>conf.auto_indent_mode</tt>.
+#
+# The default initial setting is +true+:
+#
+#   irb(main):001:0> conf.auto_indent_mode
+#   => true
+#   irb(main):002:1* Dir.entries('.').select do |entry|
+#   irb(main):003:1*   entry.start_with?('R')
+#   irb(main):004:0> end
+#   => ["README.md", "Rakefile"]
+#
+# You can change the initial setting in the
+# configuration file with:
+#
+#   IRB.conf[:AUTO_INDENT] = false
+#
+# Note that the _current_ setting <i>may not</i> be changed in the \IRB session.
+#
+# === Input \Method
+#
+# The \IRB input method determines how command input is to be read;
+# by default, the input method for a session is IRB::RelineInputMethod.
+#
+# You can set the input method by:
+#
+# - Adding to the configuration file:
+#
+#   - <tt>IRB.conf[:USE_SINGLELINE] = true</tt>
+#     or <tt>IRB.conf[:USE_MULTILINE]= false</tt>
+#     sets the input method to IRB::ReadlineInputMethod.
+#   - <tt>IRB.conf[:USE_SINGLELINE] = false</tt>
+#     or <tt>IRB.conf[:USE_MULTILINE] = true</tt>
+#     sets the input method to IRB::RelineInputMethod.
+#
+# - Giving command-line options:
+#
+#   - <tt>--singleline</tt>
+#     or <tt>--nomultiline</tt>
+#     sets the input method to IRB::ReadlineInputMethod.
+#   - <tt>--nosingleline</tt>
+#     or <tt>--multiline/tt>
+#     sets the input method to IRB::RelineInputMethod.
+#
+# See {Notes on Initialization Precedence}[rdoc-ref:IRB@Notes+on+Initialization+Precedence].
```

Comment:
I'm not a fan of putting redirection to `Initialization Precedence` all over the place. IMO the user should be aware of the different ways of setting configuration.

Reply:
I was not happy with either, but the option/config-file precedence is the opposite of what one expects in a modern app.  I'll remove the links if you think it best.

Reply:
I think it's distracting to the reader to have this line all over the place.

Reply:
Removed.

Reply:
I still see two of these, including this one.

Reply:
Removed.